Lib/asyncio/unix_events.py

```diff
@@ -1266,7 +1266,13 @@ def is_active(self):
         return True
 
     def close(self):
-        pass
+        self._join_threads()
+
+    def _join_threads(self):
+        threads = [thread for thread in list(self._threads.values())
+                            if thread.is_alive()]
+        for thread in threads:
+            thread.join()
```

Comment:
> Would it be possible to remove the thread from self._threads as soon as a thread complete?

Yeah sure, I would have no issue with adding something like this.

> Would it be possible to log a warning if a thread runs longer than timeout seconds? Or even exit silently if the timoeut is exceeded?

It would be possible, but it would likely require changing this to a coroutine, creating a new thread attached to a future that joins the threads, and using `await wait_for(future, timeout)`. This would add quite a bit of complexity. 

> Maybe we should also modify add_child_handler() to raise an exception if it's called after close() has been called? For example, add a _closed attribute? It would prevent to spawn new threads while we wait for existing threads.

Yeah this would be a fairly straightforward change and I agree with it.